src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1006,6 +1004,33 @@ static inline bool notification_match(reservation_t& res,
         ldpp_dout(res.dpp, 1) << "ERROR: failed to parse reservation id. error: " << ret << dendl;
         return ret;
       }
+    } else {
+      // no reservation - try to connect to endpoint instead
+      try {
+        const auto endpoint = RGWPubSubEndpoint::create(
```

Comment:
now that this returns a raw pointer, the memory will leak in the error paths below. please leave unique_ptr as the factory function's return type. you can move the unique_ptr into `res.endpoint` below